### ISSUE TITLE
Training Modules Suffer from Data Leakage, Scaling/Evaluation Issues, and Architecture Violations

### ISSUE THREAD

**Issue by repengine** [repengine/nautilus_trader#1]
# Bug Report

## Confirmation

- [ ] I’ve re-read the relevant sections of the documentation.
- [ ] I’ve searched existing issues and discussions to avoid duplicates.
- [ ] I’ve reviewed or skimmed the source code (or examples) to confirm the behavior is not by design.
- [ ] I’ve tested this issue using a recent *development* wheel (`dev` develop or `a` nightly) and can still reproduce it.

## Expected Behavior

- Training modules (neural, stats, XGBoost, LightGBM) should:
  - Avoid data leakage in scaling and validation.
  - Support multi-series input and configurable targets.
  - Cleanly separate model-specific logic from shared abstractions (data prep, scaling, metrics, intervals, diagnostics, monitoring).
  - Use msgspec.Struct configuration (not dicts or pydantic).
  - Provide reproducible results with central device/seed management.
  - Expose diagnostics to Prometheus and log artifacts to MLflow in a unified way.
  - Calculate prediction intervals using correct residual-based or model-native strategies.

## Actual Behavior

- Data leakage occurs in scaling: scalers are refit or reused incorrectly between train/val/test, leading to optimistic metrics and unreliable intervals.
- Single-series hardcoding prevents multi-instrument forecasting.
- Evaluation logic is inconsistent: validation alignment and horizon metrics are not robust.
- Metrics, diagnostics, and monitoring code are duplicated in each trainer module, violating DRY.
- Configurations are dict-based and lack schema validation; pydantic cannot be used due to msgspec usage in Nautilus Trader.
- MLflow and Prometheus logic are embedded within trainers, not abstracted.
- Model-specific trainers (XGBoost, LightGBM, neural, stats) contain cross-cutting concerns.

## Steps to Reproduce the Problem

1. Review `ml/training/neural_forecast.py`, `ml/training/statsforecast.py`, `ml/training/xgboost.py`, and `ml/training/lightgbm.py` for data scaling, evaluation, metrics, and diagnostics logic.
2. Attempt multi-series training or change target column—observe missing abstraction and hardcoding.
3. Try to enable diagnostics and Prometheus export—see heavy coupling inside trainers.
4. Inspect interval calculation: stats and neural modules do not use residual distributions correctly.
5. Check configuration logic: dict-based, not msgspec.Struct.
6. Attempt reproducibility: no central device/seed manager; results vary run-to-run.

## Code Snippets or Logs

```python
# Example: Leakage in neural_forecast.py
scaler.fit(train_df[features])
val_df[features] = scaler.transform(val_df[features])  # Should use train scaler only
# Single-series hardcoded:
df['unique_id'] = 'series_1'  # Fails for multi-instrument
```

## Specifications

- OS platform: Any
- Python version: 3.8+
- `nautilus_trader` version: Any
